src/parser/druid/restoration/modules/features/PrematureRejuvenations.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS/index';
+import Analyzer from 'parser/core/Analyzer';
+import HealingDone from 'parser/shared/modules/HealingDone';
+import StatisticBox from 'interface/others/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import Combatants from 'parser/shared/modules/Combatants';
+
+const debug = false;
+
+const REJUV_DURATION = 15000;
+const MS_BUFFER = 200;
+const PENDEMIC_THRESHOLD = 0.7;
+const FLOURISH_EXTENSION = 8000;
+
+/*
+ * This module tracks early refreshments of rejuvenation.
+ * TODO: Extend/refactor this module to include other HoTs/Spells as well such as lifebloom/efflorescence
+ * TODO: Add this module to checklist
+ */
+class PrematureRejuvenations extends Analyzer {
+  static dependencies = {
+    healingDone: HealingDone,
+    combatants: Combatants,
+  };
+
+  totalRejuvsCasts = 0;
+  rejuvenations = [];
+  earlyRefreshments = 0;
+  timeLost = 0;
+  pandemicExtensions = 0;
+  pandemicExtensionsTime = 0;
+
+  constructor(...args) {
+    super(...args);
+    // TODO - Extend this module to also support when using Germination.
+    this.active = !this.selectedCombatant.hasTalent(SPELLS.GERMINATION_TALENT.id);
+  }
+
+  on_byPlayer_removebuff(event) {
+    if (event.ability.guid !== SPELLS.REJUVENATION.id) {
+      return;
+    }
+    const target = this.combatants.getEntity(event);
+    if (!target) {
+      return; // target wasn't important (a pet probably)
+    }
+
+    // Sanity check - Remove rejuvenation from array if it was removed from target player.
+    this.rejuvenations = this.rejuvenations.filter(e => e.targetId !== event.targetID);
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid === SPELLS.REJUVENATION.id) {
+      this.totalRejuvsCasts++;
+
+      const oldRejuv = this.rejuvenations.find(e => e.targetId === event.targetID);
+      if (oldRejuv == null) {
+        this.rejuvenations.push({ "targetId": event.targetID, "timestamp": event.timestamp });
+        return;
+      }
+
+      const pandemicTimestamp = oldRejuv.timestamp + ((REJUV_DURATION * PENDEMIC_THRESHOLD) + MS_BUFFER);
+      if (pandemicTimestamp > event.timestamp) {
+        this.earlyRefreshments++;
+        this.timeLost += pandemicTimestamp - event.timestamp;
+      }
+
+      // Account for pandemic time if hot was applied within the pandemic timeframe.
+      let pandemicTime = 0;
+      if (event.timestamp >= pandemicTimestamp && event.timestamp <= oldRejuv.timestamp + REJUV_DURATION) {
+        pandemicTime = (oldRejuv.timestamp + REJUV_DURATION) - event.timestamp;
```

Comment:
This needs to be capped at 0.3 * rejuv duration

Reply:
yeah I misread that if statement. However that means hots that are refreshed early wont be extended by 30% (pandemic time will be 0 if it never enters that if statement).

Reply:
Missed that, thanks. Fixed now. @Gebuz